wicket/src/dispatch.rs

```diff
@@ -12,34 +12,82 @@ use clap::Parser;
 use omicron_common::{address::WICKETD_PORT, FileKv};
 use slog::Drain;
 
-use crate::{cli::ShellApp, Runner};
+use crate::{
+    cli::{CommandOutput, ShellApp},
+    Runner,
+};
 
 pub fn exec() -> Result<()> {
     let wicketd_addr =
         SocketAddrV6::new(Ipv6Addr::LOCALHOST, WICKETD_PORT, 0, 0);
 
     // SSH_ORIGINAL_COMMAND contains additional arguments, if any.
-    if let Ok(ssh_args) = std::env::var("SSH_ORIGINAL_COMMAND") {
-        // The argument is in a quoted form, so split it using Unix shell semantics.
-        let args = shell_words::split(&ssh_args).with_context(|| {
-            format!("could not parse shell arguments from input {ssh_args}")
-        })?;
-        // parse_from uses the the first argument as the command name. Insert "wicket" as
-        // the command name.
-        let app = ShellApp::parse_from(
-            std::iter::once("wicket".to_owned()).chain(args),
-        );
-
-        let log = setup_log(
-            &log_path()?,
-            WithStderr::Yes { use_color: app.global_opts.use_color() },
-        )?;
-        app.exec(log, wicketd_addr)
-    } else {
-        // Do not expose log messages via standard error since they'll show up
-        // on top of the TUI.
-        let log = setup_log(&log_path()?, WithStderr::No)?;
-        Runner::new(log, wicketd_addr).run()
+    match std::env::var("SSH_ORIGINAL_COMMAND") {
+        Ok(ssh_args) => {
+            let args = shell_words::split(&ssh_args).with_context(|| {
+                format!("could not parse shell arguments from input {ssh_args}")
+            })?;
+
+            let runtime = tokio::runtime::Runtime::new()
+                .context("creating tokio runtime")?;
+            runtime.block_on(exec_with_args(
+                wicketd_addr,
+                args,
+                OutputKind::Terminal,
+            ))
+        }
+        Err(_) => {
+            // Do not expose log messages via standard error since they'll show up
+            // on top of the TUI.
+            let log = setup_log(&log_path()?, WithStderr::No)?;
+            Runner::new(log, wicketd_addr).run()
+        }
+    }
+}
+
+/// Enables capturing of wicket's output.
+pub enum OutputKind<'a> {
+    /// Captures output to the provided log, as well as a buffer.
+    Captured {
```

Comment:
It doesn't look like this variant is ever constructed - is that coming in a subsequent PR?

Reply:
Yeah, in #4451.